docu/Concepts/TechnicalRequirements/Federation.md

```diff
@@ -41,12 +18,28 @@ To enable such a relationship between an existing community and a new community
 2. Authentication
 3. Autorized Communication
 
-### Overview
+### Overview of Federation-Handshake
 
 At first the following diagramm gives an overview of the three stages and shows the handshake between an existing community-A and a new created community-B including the data exchange for buildup such a federated, authenticated and autorized relationship.
 
 ![FederationHyperSwarm.png](./image/FederationHyperSwarm.png)
 
+### Technical Architecture
+
+The previous described handshake will be done by several technical moduls of the gradido system. The following picture gives an overview about the moduls and how the communicate with each other.
+
+![img](./image/TechnicalOverview_V1-19.svg)
+
+As soon as a Gradido Community is up and running the DHT-Modul first write the home-community-entries in the database and starts with the federation per HyperSwarm. Each community, which is configured to listen on the GRADIDO_HUB of the DHT will be part of the Gradido-Net-Federation. That means each DHT-Modul of each community will receive the publicKey of all connected communities. The DHT-Modul will open for each received publicKey a communication-socket with the associated community DHT-Modul, to exchange api-version info and hosted urls for later direct communication between both communities. The exchanged api-version info and urls will be written in the own database.
```

Comment:
```suggestion
As soon as a Gradido Community is up and running the DHT-Modul first writes the home-community-entries in the database and starts with the federation via HyperSwarm. Each community, which is configured to listen on the GRADIDO_HUB of the DHT will be part of the Gradido-Net-Federation. That means each DHT-Modul of each community will receive the publicKey of all connected communities. The DHT-Modul will open for each received publicKey a communication-socket with the associated community DHT-Modul, to exchange api-version info and hosted urls for later direct communication between both communities. The exchanged api-version info and urls will be written in the own database.
```

`GRADIDO_HUB` is variable in this case and just the default value - more correct would be `a topic` or similar description.

Event tho I know what `hosted urls` is referring to, I believe someone without knowledge about this process will have trouble to understand.

Reply:
Yes that's correct. i added a more detailed description about the background of the exchanged data api-version and url.
good hint...